### PR TITLE
[REG2.066] Issue 15733 - Forward reference issue involving inheritance

### PR DESCRIPTION
**Pull request by 9rnsr**

### CODE REVIEW

File: src/dclass.d [hunk 1]
```diff
@@ -470,20 +470,32 @@ public:
...
+             */
+            T resolveBase(T)(lazy T exp)
+            {
+                if (!scx)
+                {
+                    scx = sc.copy();
+                    scx.setNoFree();
+                }
+                _scope = scx;
+                scope (exit) _scope = null;
```
Comment (WalterBright) [inline src/dclass.d]: Using scope exit involves creating an exception handling frame. However, none of dmd throws exceptions, making this inefficient. Just set _scope to null after the call to exp().

Comment (9rnsr) [inline src/dclass.d]: OK, fixed.

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
Thanks, but the auto-merge is currently blocked by the red of Phobos unittest in stable branch.
See also: https://github.com/D-Programming-Language/phobos/pull/4092

**Pull request merged**